### DISCUSSION TITLE
Does anyone know this board?

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16068] [Q&A] [answered]
https://www.amazon.com/dp/B071KFX63R

I know it is PSF-B85 based and is here; https://tasmota.github.io/docs/Pinouts/#psf-b85psf-b01psf-b04

The problem is there are two headers but I cannot find any information on which pins are rx/tx so I can flash it.

**Comment by sfromis** [comment, marked as answer]
Those headers do not appear to be relevant to flashing. You can find the needed pins directly on the PSF-B04 module, following the pinout you already found. The tx/rx pins are not even connected to anything on the board. Here's the template you need after flashing: https://templates.blakadder.com/ewelink_4CH.html

**Reply by dac2020** [reply]
Thank you for your input.

I'm not sure I can solder directly to the chip without damaging it. I only have a 'normal' soldering item meaning, not for the tiny work these days. When looking at the image in the https://tasmota.github.io/docs/Pinouts/#psf-b85psf-b01psf-b04 link, is that showing the chip upside down or as I would see it on the board, already installed?

There is a usb connector but that seems to be for power only.

**Reply by sfromis** [reply]
You are fortunate that it is not directly to the chip which is housed inside the PSF-B04 module, as the chip is 5x5 mm with 8 pins on each edge.....

The image on the pinouts page is component side up, as you see it on the board. You can also see that by looking at the antenna placement in one corner. With stable hands and a firm grip, it may also be feasible to hand-hold pins, even if soldering is preferred for reliability. And you are right that the USB connector is solely for power.

**Reply by dac2020** [reply]
Ah yes, that's true. I'll take a look and see if I can get some leads soldered on there.

**Reply by dac2020** [reply]
I gave it a try and while the board still works, I get no communications from the rx/tx. I can only get leads onto those two as I cannot get close enough to get a third to the GND. Guess I'm done until mid Aug.

**Reply by sfromis** [reply]
You can find GND and power elsewhere on the board.

**Reply by dac2020** [reply]
I did using a pin but one of the soldered ones came off and I think it took the pad with it. 
I'll try soldering the pin back on but I suspect it's done.

Is ground needed on this one, sometimes it's not on some boards.

**Reply by sfromis** [reply]
Not sure what "this one" means, but ground the right places is as important as power. Which exact pad got torn off is important to what you can do.

**Reply by dac2020** [reply]
What I mean is that some boards only need rx/tx to communicate with while some others seem to need ground as well.
For example, I pulled my little usb to TTL device from a little router I was playing with that only needed rx/tx.

I'm not 100% sure if the pad got torn off but it's a mess to look at. Right now, the board still works but if I keep trying to re-solder leads on it, I might ruin it. I'll take a look at it later and see what's possible. I guess I have little to lose since I want something that is tasmota controlled anyhow.

**Reply by barbudor** [reply]
Tx/RX carry the signal but without GND as a level reference it won't work.
Signals are always relative to a reference.
2 circuits will need this reference even if powered by 2 different power supplies, which in that case could add some other problems

**Reply by dac2020** [reply]
Thanks. Good to know.
